tools/windows/install-help/cal/FinalizeInstall.cpp

```diff
@@ -1,5 +1,68 @@
-#include "TargetMachine.h"
 #include "stdafx.h"
+#include "PropertyReplacer.h"
+#include "TargetMachine.h"
+#include <fstream>
+
+bool updateYamlConfig(CustomActionData &customActionData)
+{
+    std::wstring inputConfig;
+
+    // Read config in memory. The config should be small enough
+    // and we control its source - so it's fine to allocate up front.
+    {
+        std::wifstream inputConfigStream(datadogyamlfile);
+
+        inputConfigStream.seekg(0, std::ios::end);
+        size_t fileSize = inputConfigStream.tellg();
+        if (fileSize <= 0)
+        {
+            WcaLog(LOGMSG_STANDARD, "ERROR: datadog.yaml file empty !");
+            return false;
+        }
+        inputConfig.reserve(fileSize);
+        inputConfigStream.seekg(0, std::ios::beg);
+
+        inputConfig.assign(std::istreambuf_iterator<wchar_t>(inputConfigStream), std::istreambuf_iterator<wchar_t>());
+    }
+
+    std::wregex re(L"^[ #]*api_key:(.*)");
```

Comment:
If I'm not mistaken, this will match any commented out API_KEY.  Not sure if this was by design or copy/paste.... but it does change the behavior.  The original code,
since comments aren't processed, would only match a non-commented API key.